py/torch_tensorrt/dynamo/conversion/impl/select.py

```diff
@@ -68,11 +69,31 @@ def select(
     indices_tensor = ctx.net.add_constant(
         index_value.shape, to_numpy(index_value)
     ).get_output(0)
-    layer = ctx.net.add_gather(input, indices_tensor, dim)
-    out = layer.get_output(0)
+    return gather(ctx, target, source_ir, name, input, dim, indices_tensor)
+
+
+def gather(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    dim: int,
+    index: Union[TRTTensor, np.ndarray, torch.Tensor],
+    sparse_grad: bool = False,
+) -> TRTTensor:
+    if not isinstance(index, TRTTensor):
+        index = get_trt_tensor(ctx, index, name + f"_parameter_to_fp32_tensor")
+    # This is for the case where torch.ops.aten.gather requires torch.int64
+    # However TRTInterpreter complains that torch.int64 is not a supported type
+    # So the below cast does not help
+    # index = cast_trt_tensor(ctx, input, trt.int32, name, target, source_ir)
```

Comment:
Does this issue still occur in the test cases?

Reply:
Yes it does. aten.scatter has similar use cases, so I am working on that. The casting of nodes in the TRT test infrastructure can be used here to get over. This is the PR- https://github.com/pytorch/TensorRT/pull/2664.